wrapper/src/main/java/software/amazon/jdbc/plugin/limitless/LimitlessConnectionPlugin.java

```diff
@@ -180,18 +181,30 @@ private void initLimitlessRouterMonitorService() {
     }
   }
 
-  private Connection retryConnection(final List<HostSpec> limitlessRouters, final Properties props, int inputRetryCount,
-      final SQLException originalException) throws SQLException {
+  private Connection retryConnectWithLeastLoadedRouters(final List<HostSpec> limitlessRouters,
+      final Properties props, final SQLException originalException) throws SQLException {
 
-    int retryCount = inputRetryCount;
+    List<HostSpec> currentRouters = limitlessRouters;
+    int retryCount = 0;
     final int maxRetries = MAX_RETRIES.getInteger(props) >= 0
```

Comment:
I believe the following line would be enough. It already includes the logic to validate user input and check for default value.

maxRetries = MAX_RETRIES.getInteger(props);